src/contracts/core/AllocationManager.sol

```diff
@@ -207,18 +214,11 @@ contract AllocationManager is
 
     /// @inheritdoc IAllocationManager
     function setAllocationDelay(address operator, uint32 delay) external {
-        require(msg.sender == address(delegation), OnlyDelegationManager());
+        // TODO: fix error
+        require(msg.sender == address(delegation) || _checkCanCall(operator, msg.sender), OnlyDelegationManager());
         _setAllocationDelay(operator, delay);
     }
 
-    /// @inheritdoc IAllocationManager
-    function setAllocationDelay(
-        uint32 delay
-    ) external {
-        require(delegation.isOperator(msg.sender), OperatorNotRegistered());
-        _setAllocationDelay(msg.sender, delay);
-    }
```

Comment:
it's unsafe to remove the check here, I think

Reply:
yup, most definitely is - added